cprover_bindings/src/goto_program/expr.rs

```diff
@@ -873,6 +874,7 @@ impl Expr {
                 (lhs.typ == rhs.typ && lhs.typ.is_integer())
                     || (lhs.typ.is_pointer() && rhs.typ.is_integer())
             }
+            ROk => lhs.typ.is_pointer() && rhs.typ.is_c_size_t(),
```

Comment:
@tautschnig

Reply:
In my first attempt I declared it as a unary operator but CBMC complained that two arguments were expected for `__CPROVER_r/w_ok`, so that is why I implemented this. I can remove if it's going to be a blocker.

Reply:
My concern had been more whether we should think of it as a binop, or as a CBMC intrinsic.  But we can merge this PR  and have a separate ticket to track that. https://github.com/model-checking/kani/issues/1111